controller/array_action/array_mediator_svc.py

```diff
@@ -254,8 +256,35 @@ def get_volume(self, volume_name, pool_id=None):
     def get_volume_name(self, volume_id):
         return self._get_volume_name_by_wwn(volume_id)
 
+    def _get_fcmaps_by_object(self, object_name):
+        all_fcmaps = []
+        fcmap_as_target = self._get_fcmap_as_target_if_exists(object_name)
+        if fcmap_as_target:
+            all_fcmaps.append(fcmap_as_target)
+        all_fcmaps.extend(self._get_fcmaps_as_source_if_exist(object_name))
+        return all_fcmaps
+
     def expand_volume(self, volume_id, required_bytes):
-        pass
+        cli_volume = self._get_cli_volume_by_wwn(volume_id, not_exist_err=True)
+        object_name = cli_volume.name
+        fcmaps = self._get_fcmaps_by_object(object_name)
+        self._safe_delete_fcmaps(object_name, fcmaps)
+        size_in_bytes = int(cli_volume.capacity)
+        expanded_bytes = required_bytes - size_in_bytes
+        try:
+            self.client.svctask.expandvdisksize(vdisk_id=cli_volume.name, unit='b', size=expanded_bytes)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if not is_warning_message(ex.my_message):
+                logger.warning("Failed to expand volume {}".format(volume_id))
+                if OBJ_NOT_FOUND in ex.my_message or VOL_NOT_FOUND in ex.my_message:
+                    raise controller_errors.ObjectNotFoundError(volume_id)
+                if NOT_ENOUGH_EXTENTS_IN_POOL_EXPAND in ex.my_message:
+                    raise controller_errors.NotEnoughSpaceInPool(pool=cli_volume.mdisk_grp_name)
+                else:
+                    raise ex
+        except Exception as ex:
+            logger.exception(ex)
+            raise ex
```

Comment:
please move this to a separate, private method

Reply:
done